File: 3194/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(minimumAverage([]int{7,8,3,4,15,13,4,1}))
}

func minimumAverage(nums []int) float64 {
	sort.Ints(nums)
	ans := 10000.0
	for i := 0; i < len(nums) / 2; i++ {
		avg := float64(nums[i] + nums[len(nums)-i-1]) / 2.0
		if avg < ans {
			ans = avg
		}
	}
	return ans
}

File: 3194/main.rs
struct Solution {}

impl Solution {
    pub fn minimum_average(nums: Vec<i32>) -> f64 {
        let mut nums = nums;
        nums.sort();
        let n = nums.len();
        let mut res = f64::INFINITY;
        for i in 0..n/2 {
            let avg = (nums[i] as f64 + nums[n-i-1] as f64) / 2.0;
            res = res.min(avg);
        }
        res
    }
}

fn main() {
    println!("{}", Solution::minimum_average(vec![1,2,3,4,5,6]));
}